### PR TITLE
Fix Realm.App constructor's TS def

### PR DESCRIPTION
**Pull request by kneth**
Closes https://github.com/realm/realm-js/issues/2940

### CODE REVIEW

File: types/app.d.ts [hunk 1]
```diff
@@ -24,9 +24,8 @@
 
 declare namespace Realm {
     // See https://stackoverflow.com/a/51114250 on why we're importing the BSON types like this
-    type ObjectId = import("bson").ObjectId;
```
Comment (kraenhansen) [inline types/app.d.ts:27]: Any reason this is getting removed? It's used by `services.d.ts` and `auth-providers.d.ts`.

Comment (kraenhansen) [reply types/app.d.ts:27]: We might want to move it to line 27 of index.d.ts if we want to use it there as well.

File: types/app.d.ts [hunk 2]
```diff
@@ -123,6 +122,9 @@ declare namespace Realm {
      * A MongoDB Realm App.
      */
     class App<FunctionsFactoryType extends object = DefaultFunctionsFactory> {
+
+        constructor(AppConstructor);
```
Comment (kraenhansen) [inline types/app.d.ts:126]: Shouldn't this be taking an `AppConfiguration` and not the `AppConstructor` type here? Did you test this manually?

**Comment by nirinchev** [comment]
What's the status of this one?

**Comment by kneth** [comment]
Superseded by #3056

**Pull request closed without merging**